tests/components/zimi/test_config_flow.py

```diff
@@ -0,0 +1,219 @@
+"""Tests for the zimi config flow."""
+
+import socket
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from zcc import ControlPointError
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.zimi.const import DOMAIN, MAC
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import format_mac
+
+
+@pytest.fixture(name="discovery_service")
+def mock_discovery_service():
+    """Mock the ControlPointDiscoveryService."""
+    with patch(
+        "homeassistant.components.zimi.config_flow.ControlPointDiscoveryService"
+    ) as service:
+        discovery = MagicMock()
+        discovery.discover = AsyncMock()
+        service.return_value = discovery
+        yield discovery
+
+
+@pytest.fixture(name="socket_mock")
+def mock_socket():
+    """Mock socket operations."""
+    with patch("homeassistant.components.zimi.config_flow.socket") as mock:
+        mock.gethostbyname = MagicMock()
+        mock.socket = MagicMock()
+        mock.AF_INET = socket.AF_INET
+        mock.SOCK_STREAM = socket.SOCK_STREAM
+        yield mock
+
+
+async def test_user_form(hass: HomeAssistant) -> None:
+    """Test we get the user form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+
+async def test_successful_config_manual(hass: HomeAssistant, socket_mock) -> None:
```

Comment:
you're missing types in your signatures

Reply:
Done https://github.com/home-assistant/core/pull/129876/commits/7bc22e9e2fb258e60be8bc538bb1235d4fb9169c